sqlglot/dialects/oracle.py

```diff
@@ -226,6 +226,41 @@ def _parse_query_restrictions(self) -> t.Optional[exp.Expression]:
                 expression=self._match(TokenType.CONSTRAINT) and self._parse_field(),
             )
 
+        def _parse_multitable_inserts(self) -> exp.MultitableInserts:
+            kind = "first" if self._prev.token_type is TokenType.FIRST else "all"
+            expressions = []
+
+            def conditional_insert():
+                expression = None
+                if self._match(TokenType.WHEN):
+                    expression = self._parse_assignment()
```

Comment:
Why is this `_parse_assignment`? I don't think we're expecting assignments in the condition, right?
```suggestion
                    expression = self._parse_disjunction()
```